### PR TITLE
[FC] Theme-based header logo

### PR DESCRIPTION
**Pull request by mats-stripe**
## Summary

This adds support for theme-based header logos in the Financial Connections flow. Right now we support Link for the `linkLight` theme, and continue to use the Stripe logo for all other themes.

## Motivation

Ultimately, the goal here is to show the Link logo in the Instant Debits flow, but is also extensible to support any other themes that might come.

## Testing

Manually verified that the link logo is only shown for the instant debits flow, and that the FC flow still shows the stripe logo. Also verified this still works with the test mode badge:

https://github.com/user-attachments/assets/f64307c8-fa14-4438-acd4-23ef0915a410

## Changelog

N/a

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift [hunk 1]
```diff
@@ -131,6 +135,10 @@ extension UIColor {
         return UIColor(red: 53 / 255.0, green: 58 / 255.0, blue: 68 / 255.0, alpha: 1)  // #353a44
     }
 
+    private static var neutral900: UIColor {
+        return UIColor(red: 23 / 255.0, green: 23 / 255.0, blue: 23 / 255.0, alpha: 1)  // ##171717
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift:139]: [Source](https://www.figma.com/design/mhgvqbODPxfR4c0j2u3ySn/%E2%9D%87%EF%B8%8F-Link-Design-System-2.0?node-id=502-41605&t=tdJA8raEm5X2pzba-4)

File: StripeFinancialConnections/StripeFinancialConnections/Resources/Images/link_logo@3x.png
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Resources/Images/link_logo@3x.png:1]: [Source](https://www.figma.com/design/mhgvqbODPxfR4c0j2u3ySn/%E2%9D%87%EF%B8%8F-Link-Design-System-2.0?node-id=934-30016&t=grVyRTje9GSGjdIG-4)

File: StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift [hunk 2]
```diff
@@ -171,6 +171,10 @@ extension UIColor {
         return UIColor(red: 26 / 255.0, green: 197 / 255.0, blue: 155 / 255.0, alpha: 1)  // #1ac59b
     }
 
+    static var linkGreen900: UIColor {
+        return UIColor(red: 1 / 255.0, green: 30 / 255.0, blue: 15 / 255.0, alpha: 1)  // #011E0F
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift:175]: [Source](https://www.figma.com/design/mhgvqbODPxfR4c0j2u3ySn/%E2%9D%87%EF%B8%8F-Link-Design-System-2.0?node-id=385-57124&t=YvBouyaJvmo0BLZ6-4)

Comment (mats-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/UIColor+Extensions.swift:175]: It's pretty subtle, but this is the colour used for the web link logo fill as well 
image

**Review by tillh-stripe — APPROVED** [review]

File: StripeFinancialConnections/StripeFinancialConnections/Source/Common/FinancialConnectionsNavigationController.swift [hunk 3]
```diff
@@ -147,12 +147,13 @@ extension FinancialConnectionsNavigationController {
     static func configureNavigationItemForNative(
         _ navigationItem: UINavigationItem?,
         closeItem: UIBarButtonItem,
-        shouldHideStripeLogo: Bool,
+        shouldHideLogo: Bool,
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Common/FinancialConnectionsNavigationController.swift:150]: good rename haha 👍

**Review by kgaidis-stripe — APPROVED** [review]

**Pull request merged**